server/src/test/java/org/elasticsearch/common/util/concurrent/PrioritizedThrottledTaskRunnerTests.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.util.concurrent;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+public class PrioritizedThrottledTaskRunnerTests extends ESTestCase {
+    private ThreadPool threadPool;
+    private Executor executor;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        threadPool = new TestThreadPool("test");
+        executor = threadPool.executor(ThreadPool.Names.GENERIC);
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        TestThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+    }
+
+    static class TestTask implements Comparable<TestTask>, Runnable {
+
+        private final Runnable runnable;
+        private final int priority;
+
+        TestTask(Runnable runnable, int priority) {
+            this.runnable = runnable;
+            this.priority = priority;
+        }
+
+        @Override
+        public int compareTo(TestTask o) {
+            return Integer.compare(priority, o.priority);
```

Comment:
Can we return `priority - o.priority` instead? `Integer.compare`normalizes to `-1, 0, 1`, thus we do not test that any other values has the right meaning.

Reply:
@henningandersen the `compareTo` is only used by the priority queue, which doesn't care for the actual distance between the two priorities, I think! So I don't understand why does this matter? Could you please elaborate why is this important?

Reply:
This is part of a test verifying that `PrioritizedThrottledTaskRunner` executes tasks in the right order when presented with *any* comparable object. But the test here is restricted to only verify using a sub-set of such comparable objects that use "normalized" return values in `compareTo`.

Arguing in terms of the implementation seems invalid, the purpose of the tests is to demonstrate that the implementation works under as many circumstances as possible. Given the simple change to demonstrate this under more circumstances here, I think we should make the change.

Reply:
I've done the change already. Thanks for the explanation! :)